### PR TITLE
csv deduplication

### PR DESCRIPTION
**Pull request by JBRibeiro**

### CODE REVIEW

File: moya/numbers.py [hunk 1]
```diff
@@ -32,8 +32,13 @@ def csv_reader(fh, batch_size=1000):
     buf = []
     for row in csv.DictReader(fh):
         number = row['to']
+        
         if not is_number(number):
-            raise Exception("Line {number} doesn't look like a phone number")
+            continue
```
Comment (mzealey) [inline moya/numbers.py:47]: keep existing logic please. if you want to make it skip instead then add a flag option and pass it in

File: moya/numbers.py [hunk 2]
```diff
@@ -32,8 +32,13 @@ def csv_reader(fh, batch_size=1000):
...
+        if number in deduplicate:
+            continue
+        else:
```
Comment (mzealey) [inline moya/numbers.py]: no need for else

File: moya/numbers.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import re
 
 NUMBER_MATCHER = re.compile(r'^[0-9]{7,15}$')
+deduplicate = set()
```
Comment (mzealey) [inline moya/numbers.py]: why are you defining this as global scope?

File: moya/numbers.py [hunk 4]
```diff
@@ -32,8 +32,13 @@ def csv_reader(fh, batch_size=1000):
     buf = []
     for row in csv.DictReader(fh):
         number = row['to']
+        
         if not is_number(number):
-            raise Exception("Line {number} doesn't look like a phone number")
+            continue
+        if number in deduplicate:
```
Comment (mzealey) [inline moya/numbers.py]: create commandline option and pass through to activate this. may not be wanted by default.

**Review by mzealey — CHANGES_REQUESTED** [review]

File: send-templated-messages.py [hunk 5]
```diff
@@ -26,7 +27,7 @@
 timer = Timer()
 
 try:
-    for items in csv_reader(args.csv):
+    for items in csv_reader(fh=args.csv, duplicate=args.deduplicate):
```
Comment (mzealey) [inline send-templated-messages.py]: very confusing naming here ? + make the deduplicate flag in the function optional defaulting to `False` as at present.

**Review by mzealey — CHANGES_REQUESTED** [review]

File: send-templated-messages.py (hunk 5 above)
Comment (JBRibeiro) [reply send-templated-messages.py]: @mzealey  Made the changes, please check the latest commit

**Review by mzealey — COMMENTED** [review]
can you also add this to all other scripts which would be reading phone numbers from files please?

**Comment by JBRibeiro** [comment]
> can you also add this to all other scripts which would be reading phone numbers from files please?

Sure, will do it

**Comment by JBRibeiro** [comment]
> can you also add this to all other scripts which would be reading phone numbers from files please?

@mzealey please check my last commit

**Review by mzealey — APPROVED** [review]

**Comment by mzealey** [comment]
good work thanks :)

**Comment by mzealey** [comment]
Had to revert this - you are adding argparse into the nested function which messes everything up. eg:

```
[root@ip-10-88-31-120 moya-public-api-examples]# python3 bulk-send.py $TOKEN test-numbers.txt ../notif.txt 
usage: bulk-send.py [-h] [--endpoint ENDPOINT] [-d] [--job-id JOB_ID] token
bulk-send.py: error: unrecognized arguments: test-numbers.txt ../notif.txt
```

See my latest commit as an eg of how to add a new argument I guess, although it's a little more complex in this case probably.

**Pull request merged**